src/Microsoft.AspNetCore.Razor.Language/DefaultRazorImportFeature.cs

```diff
@@ -0,0 +1,40 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Microsoft.AspNetCore.Razor.Language
+{
+    internal class DefaultRazorImportFeature : RazorProjectEngineFeatureBase, IRazorImportFeature
+    {
+        private IEnumerable<IRazorImportDiscoverer> _discoverers;
+
+        public IReadOnlyList<RazorSourceDocument> GetImports(RazorSourceDocument sourceDocument)
```

Comment:
I'm not sure about this working based on the source document.

Reply:
Thought about making this take a `string` but you end up losing a lot of information regarding its pathing information; especially with the added relative path. We have a separate API for getting locations on disk of imports given a string path: `IRazorImportItemFeature`

Reply:
I'm not really convinced this concept needs two levels of API/features